src/Microsoft.OData.Edm/Vocabularies/Annotations/EdmVocabularyAnnotation.cs

```diff
@@ -44,8 +48,213 @@ public EdmVocabularyAnnotation(IEdmVocabularyAnnotatable target, IEdmTerm term,
+                    foreach (var item in properties)
+                    {
+                        // get the property
+                        // get the property type from strucutral type
+                        // build the propertyvalue
+                    }
+
+                    // build the properties expression one by one
+                    return new EdmRecordExpression(....);
+                case EdmTypeKind.Collection:
+                    IEdmCollectionTypeReference collectionType = (IEdmCollectionTypeReference)typeReference;
+                    IEdmTypeReference elementType = collectionType.ElementType();
+                    // If it's in the scope:
+                    // term.DefaultValue could be a JSON string
+                    // [5, 6, 9]
+                    // [{"city":["redmond","seattle"]},{}]
+                    // Have to recursively parse the JSON array into items
+
+                    var items = ParseCollection(defaultValue);
+                    IList<IEdmExpression> itemExpressions = new List<IEdmExpression>();
+                    foreach (var item in items)
+                    {
+                        IEdmExpression itemExpr = BuildDefaultValue(elementType, item);
+                        itemExpressions.Add(itemExpr);
+                    }
+
+                    return new EdmCollectionExpression(elementType, itemExpressions);
+
+                //case EdmTypeKind.TypeDefinition:
+                //    ....
+                // Note: Need to have cases for the following types:
+                // DateTimeOffsetConstant, DecimalConstant, FloatingConstant, GuidConstant, IntegerConstant,
+                // StringConstant, DurationConstant, Null, Record, Collection, Path, If, Cast, IsType,
+                // FunctionApplication, LabeledExpressionReference, Labeled, PropertyPath, NavigationPropertyPath,
+                // DateConstant, TimeOfDayConstant, EnumMember, AnnotationPath
+            }
+        }*/
+
+        // Parse an object from a string
+        /*private static IDictionary<string, string> ParseObject(string defaultValue)
+        {
+            // parse the JSON array into items
+            // "[5, 6, 8]"  => return "5", "6", "8"
+            // defaultValue.
+            return null;
         }
 
+        // Parse a collection from a string
+        private static IEnumerable<string> ParseCollection(string defaultValue)
+        {
+            // parse the JSON array into items
+            // "[5, 6, 8]"  => return "5", "6", "8"
+            // defaultValue.
+            return null;
+        }*/
+
+        // Return an IEdmExpression for a primitive value from a string
+        /*private static IEdmExpression BuildPrimitiveValue(IEdmPrimitiveTypeReference reference, string defaultValue)
+        {
+            switch (reference.PrimitiveKind())
+            {
+                case EdmPrimitiveTypeKind.Binary:
+                    byte[] binary = new byte[0];

```

Comment:
> = new byte[0] [](http://example.com/codeflow?start=33&length=14)

shouldn't have to initialize -- can just do byte[] binary;

Reply:
Good catch! Done ✅